src/rest_api/classes/person/widgets/lineage.clj

```diff
@@ -0,0 +1,349 @@
+(ns rest-api.classes.person.widgets.lineage
+  (:require
+    [cheshire.core :as json]
+    [clojure.string :as str]
+    [datomic.api :as d]
+    [rest-api.db.main :refer [datomic-conn]]
+    [pseudoace.utils :as pace-utils]
+    [rest-api.formatters.date :as date]))
+
+(defn- generate-map [oid]
+  (let [otherpersonid (get oid 0)
+        otherpersonname (get oid 1)
+        jsondata (get oid 2)
+        roles (filter #(re-find #"role" (str %)) (keys jsondata))
+        level
+        (for [role roles
+              :let
+              [level (clojure.string/capitalize
+                       (clojure.string/replace
+                         (clojure.string/replace
+                           (str role) #":role/" "") #"-" " "))]] level)
+        duration
+        (for [role roles
+              :let [rfrom (keyword
+                            (str
+                              (clojure.string/replace 
+                                (clojure.string/replace 
+                                  (str role) #"/" ".") #":" "") "/from"))]
+              :let [rto (keyword
+                          (str
+                            (clojure.string/replace 
+                              (clojure.string/replace 
+                                (str role) #"/" ".") #":" "") "/to"))]
```

Comment:
example where this could threading macro instead:
```clojure
:let [r (-> role (str/replace #"/" ".") (str/replace #":" "")
      rto (str r "/to")]
```

Reply:
I see that it works, and it's easier to follow having the function next to the parameters, but is it better to have that extra variable as opposed to : 
                 :let [rto (keyword (str (-> role (str/replace #"/" ".") (str/replace #":" "")) "/to"))]
And in that case, should the (keyword  and (str  be in their own lines ?  I'm not sure what would be best to follow conventions and have good readability.

Reply:
Picked a way of doing the threading macros and changed the code to match.